### ISSUE TITLE
Using iTwinViewer for Connect iModels

### ISSUE THREAD

**Issue by mato5** [imodeljs/itwin-viewer#76]
Hey,

this is not really an issue but a feature-related question. There are 2 different projects/iModels that I work with, one created through the itwinjs.org portal and the other one created by iTwin Synchronizer and uploaded to connect.bentley.com portal.  However, I am unable to view the iModel from the connect.bentley.com vie the viewer-sample app provided in this repository. I've changed the iModel GUID and Context GUID in the .env file to open this iModel. When I tried to open the Project/iModel created through itwinjs.org, everything worked correctly. So my question is whether it's possible to view projects/iModels created through connect.bentley.com portal.

Best regards

**Comment by granvilj** [comment]
Hi @mato5, the viewer-sample app does have the ability to view iModels from connect.bentley.com.  To help diagnose the issue can you provide a bit more detail on what is not working.  Please provide any errors you are receiving in the console.

**Comment by mato5** [comment]
Hey @granvilj, this is the process we followed in order to get the iModel to the connect.bentley.com:
1. We created a road corridor in OpenRoads Designer
2. We used the iTwin Synchronizer app with the latest bridges to convert from DGN to iModel and upload to Connect

This is how the app looked after signing-in via the "Use Native OIDC Client" route. :
![iModeljs_error](https://user-images.githubusercontent.com/17930614/107531704-64427500-6bbd-11eb-85bc-e327faeba1c0.PNG)

And this is the error we've encountered shortly afterwards, after clicking on the Fit View button:
![iModeljs_error02](https://user-images.githubusercontent.com/17930614/107531859-8a681500-6bbd-11eb-94b3-b4a50eff7ebe.PNG)

**Comment by pmconne** [comment]
@kabentley what do you think `calculateFocusCorners` ought to be doing if `computeWorldToNpc` produces an undefined map due to non-invertible matrix? Right now it just asserts that will never happen and throws this exception when it does.

**Comment by kabentley** [comment]
> @kabentley what do you think `calculateFocusCorners` ought to be doing if `computeWorldToNpc` produces an undefined map due to non-invertible matrix? Right now it just asserts that will never happen and throws this exception when it does.

probably we should make `computeWorldToNpc` never return `undefined` . One of the extents must be zero, I'd guess. How did the view get in this state?

**Comment by calebmshafer** [comment]
@mato5 I was unable to produce the issue when testing with a couple files I have locally that I ran through the synchronizer.  Would you be able to provide the full console log and the network traffic?

Steps are [here for logs](https://support.shortpoint.com/support/solutions/articles/1000222881-save-browser-s-console-file#Step-3%3A-Save-the-logs) and [here for network](https://community.jaspersoft.com/wiki/exporting-firebug-and-chrome-developer-tool-net-panel-http-archive-file#:~:text=In%20Chrome%2C%20the%20ability%20to,panel%20to%20the%20file%20system.) if you haven't done it before.  

Also, you may want to grep for `"name": "authorization"` in the resulting har file from the network info and remove all of the values with a Bearer token before attaching it here.

**Comment by kckst8** [comment]
@mato5 , can you please try to use the latest version of the iTwin Viewer React package (6.0.4) in your app and let us know if it resolves your issue?

**Comment by mato5** [comment]
We've just updated the iTwin Viewer package as the @kckst8 suggested and tested it with 2 different iModels. First iModel was the one we used earlier and the issue persisted. The second iModel was a different one and it worked correctly. It works with iModel created in OpenRail Designer CE 10.09 with "Metric Standards" Workspace delivered with program installation.
Problem persists with iModel created in OpenRoads Designer CE 10.09 with "CZ Standards" Workspace active and with DGN created in local Slovak coordinates, which means that all model is located in III. quadrant (-x, -y).

Could this be the root of our problem?
